includes/core/classes/class-event-setup.php

```diff
@@ -74,6 +75,151 @@ protected function setup_hooks(): void {
 		add_filter( 'get_the_date', array( $this, 'get_the_event_date' ) );
 		add_filter( 'the_time', array( $this, 'get_the_event_date' ) );
 		add_filter( 'display_post_states', array( $this, 'set_event_archive_labels' ), 10, 2 );
+
+		/**
+		 * WHERE SHOULD ALL THAT GO?
+		 *
+		 * @todo Refactor into a
+		 *       class-add-to-calendar.php
+		 *       or another DRY place ?!
+		 */
+		add_action(
+			'init',
+			function () {
+				// $settings            = Settings::get_instance();
+				// $event_rewrite_base = $settings->get_value( 'general', 'urls', 'events' );
+				$event_rewrite_base = 'event'; // Replace when #812 is merged.
+				$calendar_endpoints = join(
+					'|',
+					array(
+						'googlecalendar',
+						'yahoocalendar',
+						'ical',
+						'iCal',
+						'outlook',
+					)
+				);
+
+				/**
+				 * Generate rewrite rules for endpoints like:
+				 * event/my-sample-event/(ical|iCal|googlecalendar)(/)
+				 */
+				add_rewrite_rule(
+					sprintf(
+						'%s/([^/]+)/(%s)/?$',
+						$event_rewrite_base,
+						$calendar_endpoints
+					),
+					'index.php?gatherpress_event=$matches[1]&gatherpress_ext_calendar=$matches[2]',
+					'top'
+				);
+
+				/**
+				 * Filters the query variables allowed before processing.
+				 *
+				 * @param string[] $public_query_vars The array of allowed query variable names.
+				 * @return string[] The array of allowed query variable names.
+				 */
+				add_filter(
+					'query_vars',
+					function ( array $public_query_vars ): array {
```

Comment:
Let's avoid anonymous functions in hooks. They are quite hard to debug and test.